### PR TITLE
Refactor: Eliminate Global Variable to Enhance Testability

### LINKED ISSUE #4217: chore: remove global variable for better unit testing
In Litmus version 2, we already remove global variables. you can see here #3937 

However, with the change of Litmus version to 3, we have a new global variable, which we need to change to a field in the struct because global variables make unit testing difficult.

Here's a global variable what we need to remove.
https://github.com/litmuschaos/litmus/blob/794676f6b75e08835334fd5d675e0b0d096d5f9d/chaoscenter/graphql/server/pkg/database/mongodb/operations.go#L36

We need to find and change all the places where that global variable is used, like this
Screenshot 2023-10-05 at 11 37 03 AM

Here's an example of what we did first #4136 

Anyone who finds good first issues or participates in hacktoberfest can take this issue.

### PR DESCRIPTION
**Pull request by andoriyaprashant**
## Proposed changes

This pull request refactors the codebase by removing a global variable to improve testability.

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by andoriyaprashant** [comment]
Hello @namkyu1999 Please review my pull request

**Comment by namkyu1999** [comment]
can you check DCO?

**Comment by andoriyaprashant** [comment]
@namkyu1999 Sir, as you instructed, I've tried to complete the checks. Could you please verify them? If there are any changes needed in the code, please let me know, and I'll try to fix them

**Comment by namkyu1999** [comment]
Hi @andoriyaprashant, some checks are failing now (build-pipeline, CIFuzz). can you check them?

**Comment by andoriyaprashant** [comment]
> Hi @andoriyaprashant, some checks are failing now (build-pipeline, CIFuzz). can you check them?

i am working on those 👍🏻

File: chaoscenter/graphql/server/pkg/database/mongodb/operations.go [hunk 1]
```diff
@@ -32,8 +32,8 @@ type MongoOperations struct {
 }
 
 var (
```
Comment (Jonsy13) [inline chaoscenter/graphql/server/pkg/database/mongodb/operations.go:34]: Can we remove this var block itself as well.

File: chaoscenter/graphql/server/pkg/database/mongodb/config/operations.go [hunk 2]
```diff
@@ -10,20 +10,20 @@ import (
 )
 
 // CreateConfig creates a new server config with unique key
-func CreateConfig(ctx context.Context, config *ServerConfig) error {
-	err := mongodb.Operator.Create(ctx, mongodb.ServerConfigCollection, config)
+func CreateConfig(ctx context.Context, config *ServerConfig, mongodbOperator mongodb.MongoOperator) error {
```
Comment (Saranya-jena) [inline chaoscenter/graphql/server/pkg/database/mongodb/config/operations.go:13]: Is it possible to create an interface and receivers here similar to what we have in other places?

**Comment by andoriyaprashant** [comment]
Hello @Saranya-jena @Jonsy13 @namkyu1999 
I made some changes to the files as @Jonsy13  suggested removing the var() block, so I removed it completely. Also, as per @Saranya-jena  suggestion, I made some changes to the config file. Please review them, and if there are any errors or further changes needed, please let me know. I'll do my best to fix them.
Thank you

**Review by Saranya-jena — APPROVED** [review]

**Comment by Saranya-jena** [comment]
The checks are failing can you fix those?

**Comment by Saranya-jena** [comment]
@andoriyaprashant are you still working on this PR?

**Comment by andoriyaprashant** [comment]
> @andoriyaprashant are you still working on this PR?

Sorry @Saranya-jena I forgot to check the pull request checks. I'll fix them as soon as possible 👍🏻

**Comment by Saranya-jena** [comment]
@andoriyaprashant the checks are still failing, PTAL.

**Comment by andoriyaprashant** [comment]
> @andoriyaprashant the checks are still failing, PTAL.

Ma'am @Saranya-jena   , can you please help me or guide me in fixing the CIFuzz workflow error? I tried, but I am not able to understand it

**Comment by S-ayanide** [comment]
Can you please rebase with main? The current commit history is incorrect which is why it's pulling a lot of extra commits too.

**Comment by Saranya-jena** [comment]
@andoriyaprashant I think the branch history of this PR got messed up, that's why you can see the checks are failing and the commits of other contributors are also added. Can you raise a new PR for these changes?

**Comment by andoriyaprashant** [comment]
> @andoriyaprashant I think the branch history of this PR got messed up, that's why you can see the checks are failing and the commits of other contributors are also added. Can you raise a new PR for these changes?

Sure

**Pull request closed without merging**